### PR TITLE
Handle empty string columns in slice reader

### PR DESCRIPTION
**Pull request by bhhari**
```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDirectSelectiveStreamReader.java [hunk 1]
```diff
@@ -328,7 +332,10 @@ private void skipData(int start, int items)
                 lengthIndex++;
             }
         }
-        dataStream.skip(dataToSkip);
+        //in case of an empty varbinary both the presentStream and dataStream are null and only lengthStream is present.
```
Comment (highker) [inline presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDirectSelectiveStreamReader.java]: Add a space after //

File: presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDirectSelectiveStreamReader.java [hunk 2]
```diff
@@ -306,15 +306,19 @@ else if (nullsAllowed) {
     private void skip(int items)
             throws IOException
     {
-        if (dataStream == null) {
+        //in case of an empty varbinary both the presentStream and dataStream are null and only lengthStream is present.
```
Comment (highker) [inline presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDirectSelectiveStreamReader.java]: Add a space after //

**Comment by bhhari** [comment]
@oerling  had a look but some reason his email reply did not come in. @highker  this is good for merge.

File: presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDirectSelectiveStreamReader.java [hunk 3]
```diff
@@ -306,15 +306,19 @@ else if (nullsAllowed) {
...
-        if (dataStream == null) {
+        // in case of an empty varbinary both the presentStream and dataStream are null and only lengthStream is present.
+        if (dataStream == null && presentStream != null) {
             presentStream.skip(items);
         }
         else if (presentStream != null) {
             int lengthToSkip = presentStream.countBitsSet(items);
             dataStream.skip(lengthStream.sum(lengthToSkip));
         }
         else {
```
Comment (highker) [inline presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDirectSelectiveStreamReader.java]: nit: 
```java
else if (dataStream != null) {
    dataStream.skip(lengthStream.sum(items));
}

Comment (bhhari) [reply presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDirectSelectiveStreamReader.java]: @highker  in that case lengthStream will not be proceed right

Comment (highker) [reply presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDirectSelectiveStreamReader.java]: I'm not sure actually. The snippet I put should be semantically equivalent as yours, isn't it?

Comment (bhhari) [reply presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDirectSelectiveStreamReader.java]: lengthStream.sum advances the currentPosition.
lets say we need to skip 10 positions then lengthStream.sum(10) will return the number of data items to skip and at the same time skips 10 positions in the lengthStream, but the snippet you have does not enter inside the codition as dataStream is null.

Comment (highker) [reply presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDirectSelectiveStreamReader.java]: Got it. I didn't see `sum` has side effect.

**Pull request merged**